### DISCUSSION TITLE
Add a new AsymmetricRelatedField to DRF

### DISCUSSION THREAD

**Discussion by saadullahaleem** [encode/django-rest-framework#8966] [Ideas & Suggestions] [answered]
Hey folks. I've been using DRF for a while and one of the neat things I picked up a few years ago is a `AsymmetricRelatedField`. Here's the sample code:

```python
class AsymmetricRelatedField(serializers.PrimaryKeyRelatedField):
    def __init__(
        self, serializer_class: Any, *args: Any, **kwargs: Any
    ) -> None:
        self.serializer_class = serializer_class
        super().__init__(*args, **kwargs)

    def to_representation(self, value: Any) -> Any:
        return self.serializer_class(value).data


class MySerializer:
    destination = AsymmetricRelatedField(
        CitySerializer, queryset=City.objects.all()
    )
```

This should be pretty self-explanatory. it simply lets you use the same field for accepting a Primary key and returns an object based on a serializer in the response. Would it be a good idea to have this within DRF or is this something that's too basic? 

If I were to implement this as core functionality, I would add the `serializer_class` to the constructor of the `RelatedField` that all these fields inherit from. This way, if the `serializer_class` is present, the `to_representation` method of `RelatedField` would use is `serializer_class`.  

The `to_representation` methods of the fields that inherit from `RelatedField` would then first check if the serializer class is present or not via calling the base class's `to_representation` before falling back to their own implementations.

If this sounds like a good idea, I'd love to open my first PR for it!

**Comment by auvipy** [comment, marked as answer]
At this point of this projects life cycle, it's highly unlikely to add new feature to DRF core unless it is supported in django. so I request to send a PR here https://github.com/chibisov/drf-extensions and ping me
